messages/jsonschema/scripts/templates/java.java.erb

```diff
@@ -0,0 +1,111 @@
+<% @schemas.each do |key, schema| %>
+    public static class <%= class_name(key) %> {
+      <%- schema['properties'].each do |property_name, property| -%>
+        private <%= type_for(class_name(key), property_name, property) -%> <%= property_name %>;
+      <%- end -%>
+
+      <%- if (schema['required'] || []).empty? -%>
+        public <%= class_name(key) %>() {}
+      <%- schema['properties'].each_with_index do |(property_name, property), index| -%>
+
+        public static <%= class_name(key) %> from(<%= type_for(class_name(key), property_name, property) -%> <%= property_name %>) {
+          <%= class_name(key) %> o = new <%= class_name(key) %>();
+          <%- if property['items'] -%>
+          o.<%= property_name %> = java.util.Collections.unmodifiableList(new java.util.ArrayList<>(java.util.Objects.requireNonNull(<%= property_name %>, "<%= class_name(key) %>.<%= property_name %> cannot be null")));
+          <%- else -%>
+          o.<%= property_name %> = java.util.Objects.requireNonNull(<%= property_name %>, "<%= class_name(key) %>.<%= property_name %> cannot be null");
+          <%- end -%>
+          o.validate();
+          return o;
+        }
+      <%- end -%>
+      <%- else -%>
+        private <%= class_name(key) %>() {}
+      <%- end -%>
+
+        public <%= class_name(key) %>(
+        <%- schema['properties'].each_with_index do |(property_name, property), index| -%>
+            <%= type_for(class_name(key), property_name, property) -%> <%= property_name %><%= index < schema['properties'].length-1 ? ',' : ''%>
+        <%- end -%>
+        ) {
+          <%- schema['properties'].each_with_index do |(property_name, property), index| -%>
+            <%- if property['items'] -%>
+            this.<%= property_name %> = <%= property_name %> == null ? null : java.util.Collections.unmodifiableList(new java.util.ArrayList<>(<%= property_name %>));
+            <%- else -%>
+            this.<%= property_name %> = <%= property_name %>;
+            <%- end -%>
+          <%- end -%>
+            validate();
+        }
+
+      <%- schema['properties'].each do |property_name, property| -%>
+        <%- if (schema['required'] || []).index(property_name) -%>
+
+        public <%= type_for(class_name(key), property_name, property) -%> get<%= capitalize(property_name) %>() {
+            return <%= property_name %>;
+        }
+        <%- else -%>
+
+        public java.util.Optional<<%= type_for(class_name(key), property_name, property) -%>> get<%= capitalize(property_name) %>() {
+            return java.util.Optional.ofNullable(<%= property_name %>);
+        }
+        <%- end -%>
+      <%- end -%>
+
+        private void validate() {
```

Comment:
Don't forget to remove this.

Reply:
If the constructors already have the `requireNonNull` for each required argument then what is there left to validate?